File: src/main/java/core/db/test/Lesson.java
package core.db.test;

import core.db.annotation.Entity;
import core.db.annotation.Id;
import core.db.annotation.Property;
import core.db.annotation.ToOne;

@Entity(table = "lesson")
public class Lesson {

    @Id(autoincrement = true, name = "id")
    private long id;

    @Property(name = "title")
    private String title;

    @Property(name = "description")
    private String description;

    @ToOne(joinedEntity = Student.class)
    private Student student;

}

File: src/main/java/core/db/test/Lesson.scala
package core.db.test

import com.j256.ormlite.field.{DatabaseField => Column}
import com.j256.ormlite.table.{DatabaseTable => Table}

object Lesson {
  def apply(): Lesson = new Lesson()
}

@Table(tableName = "lesson")
final class Lesson private() {

  @Column(generatedId = true)
  var id: Int = _

  @Column(columnName = "title")
  var title: String = _

  @Column(columnName = "description")
  var description: String = _

  @Column(columnName = "user",
    foreign = true,
    foreignAutoRefresh = true)
  var student: Student = _

  override def toString: String = s"""id=$id, title=$title, description=$description, student=${student.id}"""

}
